File: Part1_강의자료/Part1_쪽지시험/문제별코드/ProblemC_먹을것인가먹힐것인가/7795.py
T = int(input())

for _ in range(T):
    N, M = map(int, input().split())

    a = list(map(int, input().split()))
    b = list(map(int, input().split()))

    a.sort()
    b.sort()

    ans = 0
    bi = 0
    for i in range(N):
        while bi < M and b[bi] < a[i]:
            bi += 1
        ans += bi
    print(ans)

File: Part1_강의자료/Part1_쪽지시험/문제별코드/ProblemC_먹을것인가먹힐것인가/7795.cpp
#include <stdio.h>
#include <vector>
#include <algorithm>
using namespace std;

int main()
{
    int T;

    scanf("%d", &T);
    while (T--)
    {
        int N, M;
        scanf("%d %d", &N, &M);

        vector<int> a(N);
        for (int i = 0; i < N; i++)
            scanf("%d", &a[i]);
        
        vector<int> b(M);
        for (int i = 0; i < M; i++)
            scanf("%d", &b[i]);
        
        sort(a.begin(), a.end());
        sort(b.begin(), b.end());

        int ans = 0;
        int bi = 0;
        for (int i = 0; i < N; i++)
        {
            while (bi < M && b[bi] < a[i])
                bi++;
            ans += bi;
        }
        printf("%d\n", ans);
    }
    return 0;
}
